doc/dao/Managing Git & Formatting PRs.md

```diff
@@ -0,0 +1,53 @@
+# Managing Git for Primero
+
+As part of the process for contributing to Primero, it is necessary to maintain a specific workflow for using Git and GitHub.
+
+You should initially create a fork of Primero on GitHub. If you are working on Primero as part of an organization, there might already be an existing fork on GitHub.
+
+![Forking on github!](Attachments/fork.webp)
+Once you have created a fork, you should add your fork as a remote destination to your local copy of the primero code.
+
+Initially, you should create a branch with a descriptive name which identifies the feature you are working on, and push the branch to your fork of Primero hosted on GitHub.
+
+```bash
+git checkout develop
+git pull
+git remote add fork https://github.com/<YOUR GITHUB USERNAME>/primero.git
+git fetch fork
+git checkout -b my-branch-name
+git push --set-upstream fork my-branch-name
+```
+
+## Creating a Pull Request
+
+Once you are ready to have your contribution reviewed by the core development team, you should open a Pull Request on the Primero repository. The development team will then review the contribution, and suggest changes. Once you implement those changes, the team will review the contribution again, and either the review process will continue, or the changes will be approved for merging. Once the changes are merged, and the contribution is concluded, you will be able to ask for payment from the DAO.
+
+If your contribution is not ready, but you need a place to discuss the changes you are making, you can make a pull request, and mark it as a draft.
+
+# Pull Request Checklist
+
+If all of the following are not true, it is unlikely that the proposed PR will be accepted. With prior agreement from the Primero development team, it is possible to waive some of these requirements, but this agreement should be sought before embarking on a new feature.
+
+Even if all of the requirements in the checklist are met, the code is still subject to in-depth review by the Primero development team, who hold ultimate authority regarding what code is merged. You should expect that the development team will suggest changes multiple times during the code review process.
+## Backend Checklist
+- [ ] RuboCop is passing without warnings on all changed Ruby files
+- [ ] No new Ruby dependencies have been added to Gemfile
+- [ ] Solr is not used within any new code
+- [ ] Unit tests exist and pass for any new Ruby code
+- [ ] No changes result in N+1 calls to the database
+- [ ] There are no outstanding performance issues with the backend code
+## Frontend Checklist
+- [ ] ESLint is passing without any warnings on all changed JavaScript files
+- [ ] No new JavaScript dependencies have been added to package.json
```

Comment:
We should also mention no new deps unless it's mentioned in the proposal. There might be some instances where a new package should be added for example a custom calendar for a newly added language. Also add some text on the proposal page

Reply:
Same for additions to the gemfile

Reply:
Agree on JS with @jtoliver-quoin. However I feel a bit more draconian about the Gemfile. Ruby dependencies are likelier to be stale these days and Rails does like 95% of what we need so I would prefer to keep things that aren't maintained by the RoR team or Shopify or other big contributors unless explicitly agreed to.

Reply:
I've added some language to the proposal section which probably helps with this